mcp/mcp-cli/src/main/java/software/amazon/smithy/java/mcp/cli/commands/StartServer.java

```diff
@@ -151,20 +157,28 @@ public void execute(ExecutionContext context) throws IOException {
 
         ThrowingRunnable awaitCompletion;
         Supplier<CompletableFuture<Void>> shutdownMethod;
+        Set<String> allowedTools = null;
         if (proxyServer != null) {
             proxyServer.start();
             awaitCompletion = proxyServer::awaitCompletion;
             shutdownMethod = proxyServer::shutdown;
         } else {
             if (registryServer) {
-                services.add(McpRegistry.builder()
-                        .addInstallServerOperation(new InstallOp(registry, config))
-                        .addListServersOperation(new ListOp(registry))
-                        .build());
+                allowedTools = new CopyOnWriteArraySet<>();
```

Comment:
Does this need to be initialized with install and search?

Reply:
Nice catch! I had added in my local testing. 

This also led me to realize that servers passed along with `--registry` also need to be allowlisted.